### DISCUSSION TITLE
Create a function where you type a text and this text is transformed into .mp3 file

### DISCUSSION THREAD

**Discussion by Marcioghiraldelli** [AzuraCast/AzuraCast#4288] [Ideas]
It would be very interesting if it had a function where you could type a text, choose the language and come out as a voice on the radio.
Example: hello listeners
the system would take this text and turn it into a voice with someone talking, like those google assistants and go to the radio.

**Comment by learning-more** [comment]
@Marcioghiraldelli Agreed, this would be very interesting! This would be a perfect example of writing a plugin. More plugin details are here: https://docs.azuracast.com/en/developers/plugins

Azuracast uses a tool called LiquidSoap to create the AutoDJ playlists. LiquidSoap can do more than just playlists, including creating audio files that contain speech. There is a [list of options here](https://www.liquidsoap.info/doc-dev/protocols.html). Specifically, LiquidSoap can use Google's Text to Speech engine, Amazon POLLY, TEXT2WAVE and other text to speech engines.

Once you have an audio file, LiquidSoap offers an "append track" option, [noted in the Azuracast docs here](https://docs.azuracast.com/en/developers/liquidsoap). 

I'm sure if you paid to have a plugin developed, many stations would use it!

**Reply by Marcioghiraldelli** [reply]
yes, I'm very interested in the plugin, but I don't think I can make or develop it, so I informed here if it would be possible to add.

**Reply by Marcioghiraldelli** [reply]
f you know someone who does this, it would be nice, I would pay depending on the amount of course and I would make it available to everyone who uses azuracast.
It would be a way for me to contribute to the project

**Reply by nwithan8** [reply]
I like the idea. I might take a crack at it, but I've never done PHP development before.
